enterprise/server/cmd/ci_runner/main.go

```
@@ -1646,6 +1649,11 @@ func (ws *workspace) sync(ctx context.Context) error {
 		return err
 	}
 	merge := action.GetTriggers().GetPullRequestTrigger().GetMergeWithBase()
+	if *gitFetchDepth > 0 {
+		// If not fetching the full commit history, then we might not have the
+		// merge-base commit available, so we can't reliably merge.
+		merge = false
```

Instead of silently turning off merging, is it possible to error out here?